data-analytics/dataflow-python-examples/dataflow_python_examples/data_generation_for_benchmarking.py

```diff
@@ -0,0 +1,485 @@
+            'address': 'address',
+            'file': 'file_name',
+            'color': 'color_name',
+            'zip': 'zipcode',
+            'phone': 'phone_number',
+            'name': 'name',
+            'company': 'company',
+            'month': 'month',
+            'city': 'city',
+            'state': 'state',
+            'country': 'country',
+            'username': 'user_name',
+            'email': 'email',
+            'num': 'random_number',
+            'description': 'paragraph'
+        }
+
+        faker_schema = {}
+        for obj in self.schema:
+            is_special = False
+            for key in special_map:
+                if key.lower() in obj['name'].lower():
+                    faker_schema[obj['name']] = special_map[key]
+                    is_special = True
+                    break
+            if not is_special:
+                faker_schema[obj['name']] = type_map[obj['type']]
+
+        return faker_schema
+
+
+class FakeRowGen(beam.DoFn):
+    """
+    This class wraps the logic defined in DataGenerator object and generates a fake record for each
+    element it is passed.
+    """
+    def __init__(self, data_gen):
+        """
+        This initiates some properties of the FakeRowGen DoFn including an instance of the
+        DataGenerator class and the number of records should be generated for each element in the
+        prior PCollection.
+
+        Attributes:
+            data_gen: A DataGenerator object defining the shape of the data should be generated by
+                      this DoFn.
+            n: An integer specifying how many records to generate.
+        """
+        self.data_gen = data_gen
+
+    # Helper function to get a single field dictionary from the schema for checking type and mode.
+
+    def get_field_dict(self, field_name):
+        return filter(lambda f: f[u'name'] == field_name,
+                      self.data_gen.schema)[0]
+
+    def sanity_check(self, record, fieldname):
+        """
+        This function ensures that the data is all of types that BigQuery expects.
+
+        Args:
+            record: A dictionary generated by faker_schema representing the candidate for a fake
+                    row in our BigQuery table
+            fieldname: name of field we are checking with this call.
+        """
+        # Create a Faker instance for individual parameterized random generation
+        # (ie. minimum date).
+        faker = Faker()
+
+        field = self.get_field_dict(fieldname)
+        # Below handles if the datatype got changed by the faker provider
+        if field[u'type'] == 'STRING' or field[u'type'].find('TIME') > -1:
+            record[fieldname] = unicode(record[fieldname])
+        elif field[u'type'].find('DATE') > -1:
+            # This implements the minimum/maximum date functionality
+            # and avoids regenerating a random date if already obeys min/max date.
+            if record[fieldname] < self.data_gen.min_date \
```

Comment:
Style guide requires the use of parens instead of \ for line splitting:

```
elif (record[fieldname < self.data_gen.min_date or
    record[fieldname] > self.data_gen.max_date):
```

Reply:
Correcting styling.